src/types-and-traits.rst

```diff
@@ -2048,6 +2048,19 @@ occur when:
 * :dp:`fls_SYnFJBhi0IWj`
   The source :t:`type` is a :t:`trait object type` and the target :t:`type` is a :t:`trait object type` with the same or no :t:`[principal trait bound]`, and the target :t:`type` has the same or less non-:t:`principal trait` :t:`[trait bound]s`.
 
+* :dp:`fls_QB4c6FNKxaPl`
+  The source :t:`type` is ``dyn T`` and the target :t:`type` is ``dyn U``,
+  where ``U`` is a :t:`supertrait` of ``T``.
+
+  .. code-block:: rust
+
+     trait Sub: Super {}
+     trait Super {}
+
+     fn upcast(x: &dyn Sub) -> &dyn Super {
+         x // implicit coercion
+     }
+
```

Comment:
```suggestion
* :dp:`fls_QB4c6FNKxaPl`
  The source :t:`type` is a :t:`trait object type` with some :t:`principal trait bound` ``T`` and the target :t:`type` is a :t:`trait object type` with some :t:`principal trait bound` ``U``, where ``U`` is a :t:`supertrait` of ``T``.

```

Note that we don't throw in code example blocks arbitrarily unless describing the structure is fairly complicated (and even then I feel like it is wrong to do so)